core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java

```diff
@@ -92,57 +96,92 @@ public View clone() {
+        }
 
-            return this;
+        @JsonIgnore
+        public CHILD_CLASS json(final byte[]... jsonBytes) throws SchemaException {
+            return json(View.class, jsonBytes);
         }
 
         @Override
-        public Builder entity(final String group, final ViewElementDefinition elementDef) {
-            return (Builder) super.entity(group, elementDef);
-        }
+        @JsonIgnore
+        public CHILD_CLASS merge(final View view) {
+            if (getThisView().entities.isEmpty()) {
+                getThisView().entities.putAll(view.getEntities());
+            } else {
+                for (final Map.Entry<String, ViewElementDefinition> entry : view.getEntities().entrySet()) {
+                    if (!getThisView().entities.containsKey(entry.getKey())) {
+                        entity(entry.getKey(), entry.getValue());
+                    } else {
+                        final ViewElementDefinition mergedElementDef = new ViewElementDefinition.Builder()
+                                .merge(getThisView().entities.get(entry.getKey()))
+                                .merge(entry.getValue())
+                                .build();
+                        getThisView().entities.put(entry.getKey(), mergedElementDef);
+                    }
+                }
+            }
 
-        public Builder entity(final String group) {
-            return entity(group, new ViewElementDefinition());
+            if (getThisView().edges.isEmpty()) {
+                getThisView().edges.putAll(view.getEdges());
+            } else {
+                for (final Map.Entry<String, ViewElementDefinition> entry : view.getEdges().entrySet()) {
+                    if (!getThisView().edges.containsKey(entry.getKey())) {
+                        edge(entry.getKey(), entry.getValue());
+                    } else {
+                        final ViewElementDefinition mergedElementDef = new ViewElementDefinition.Builder()
+                                .merge(getThisView().edges.get(entry.getKey()))
+                                .merge(entry.getValue())
+                                .build();
+                        getThisView().edges.put(entry.getKey(), mergedElementDef);
+                    }
+                }
+            }
+
+            return self();
         }
 
-        public Builder entities(final Collection<String> groups) {
-            for (final String group : groups) {
-                entity(group);
-            }
+        private View getThisView() {
+            return getElementDefs();
+        }
+    }
 
-            return this;
+    @JsonPOJOBuilder(buildMethodName = "build", withPrefix = "")
+    public static final class Builder extends BaseBuilder<Builder> {
+        public Builder() {
         }
 
-        @Override
-        public View build() {
-            return (View) super.build();
+        public Builder(final View view) {
+            this();
```

Comment:
Sorry, I mean the call onto this() (the no-arg constructor) which doesn't contain a body.